compiler/rustc_codegen_gcc/src/lib.rs

```diff
@@ -27,7 +27,6 @@
 #![allow(clippy::needless_lifetimes, clippy::uninlined_format_args)]
 
 // Some "regular" crates we want to share with rustc
-extern crate object;
 extern crate smallvec;
```

Comment:
So, we apparently run the risk of eventually having the same problem with smallvec. Should we just make this a crates.io dependency as well?

Reply:
Can do - what's the policy here? Should only `rustc_*` crates be pulled from sysroot?

Reply:
I don't think we have a policy, we just discovered this problem.^^

`tracing` needs to be pulled from the sysroot since it manages a singleton. Other than that, the main thing is avoiding duplicates... but also, if there's a `SmallVec` in a rustc type, then ofc it needs to use the sysroot version of the crate.

We can also just wait if one of the other crates becomes a problem. But it's going to be hard to notice that if we don't even build this on CI. So maybe the more important thing is to ensure these crates get check-built on more targets. @Kobzol might have an idea how to do that?

Reply:
I used to pull as much as I could from the sysroot to save on compile time, but it unfortunately, we have to stop doing this for Windows it seems.

Reply:
@antoyo this is not a windows-specific problem in general, it just so happens that in the windows build, the compiler doesn't load `object` beforehand, and there are multiple candidates.

That's why I asked about a policy - we need to say what crates are guaranteed to exist exactly once in the sysroot, as these can be pulled in safely, and no others.

Reply:
It's hard to say which ones are *guaranteed* to exist only once. OTOH, this in principle affects all rustc drivers (clippy, Miri, rustdoc) and hardly ever seems to be a problem... but also, at least in Miri we hardly use `extern crate`, so maybe that's why.

Reply:
> We can also just wait if one of the other crates becomes a problem. But it's going to be hard to notice that if we don't even build this on CI. So maybe the more important thing is to ensure these crates get check-built on more targets. @Kobzol might have an idea how to do that?

We don't *check* almost anything on CI, but we could build the GCC backend on more OSes. We haven't done that so far because it's not even in blocking (auto) CI on Linux yet, but that's for tests. For just building the backend, I guess we could try to add it to more runners.

Reply:
Building might be harder than checking and require a full GCC... but maybe it can be done. Just seems like overkill when the main goal is to allow contributors to fix compilation failures.

Reply:
I think it would make sense to do any further changes like re-exporting smallvec as a separate PR, so would be great if I could get an approval on this.